launcher/net/NetJob.cpp

```diff
@@ -145,10 +146,22 @@ void NetJob::updateState()
                   .arg(QString::number(m_doing.count()), QString::number(m_done.count()), QString::number(totalSize())));
 }
 
+bool NetJob::isOnline()
+{
+    // check some errors that are ussually associated with the lack of internet
+    for (auto job : getFailedActions()) {
+        auto err = job->error();
+        if (err != QNetworkReply::HostNotFoundError && err != QNetworkReply::NetworkSessionFailedError) {
+            return true;
+        }
+    }
+    return false;
```

Comment:
This logic returns true on the first error that's not an internet error and ignores the case of no errors
```suggestion
    for (auto job : getFailedActions()) {
        auto err = job->error();
        if (err == QNetworkReply::HostNotFoundError || err == QNetworkReply::NetworkSessionFailedError) {
            return false;
        }
    }
    return true;
```

Reply:
Yeah, and that is intended: if another error is not related to the internet being down then consider that the connection is online( NoError means that one request succeded and the instance is connected to the internet).
Only in case, all errors indicate no internet then do not display the dialog and consider it offline.

Reply:
ah, I we misreading the use.